### PR TITLE
[RHOSINFRA-70] Image settings

### PR DESCRIPTION
**Pull request by obaranov**
Replace the image* arguments for provisioner with distro one. 

@tkammer, @yfried-redhat please review. 

This change requires also CI private change: https://gitlab.cee.redhat.com/rhosqeauto/InfraRedCI/merge_requests/15

### CODE REVIEW

**Comment by obaranov** [comment]
POC job: http://10.8.180.145:8080/job/poc-ir-obaranov-distro/4/console

File: settings/provisioner/distro/rhel-7.2.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+image:
+  file: rhel-guest-image-7.2-20160302.0.x86_64.qcow2
+  server: "{{ !lookup private.images.rhel.7.2.remote_file_server }}"
+name: rhel
```
Comment (yfried-redhat) [inline settings/provisioner/distro/rhel-7.2.yml]: Why do we need this data?
We are using images to specify the distro, so we don't need to hold it in yaml

```yaml
name: rhel
full_version: "7.2"
 version:
   major: '7'
   minor: '2'
```

Comment (obaranov) [inline settings/provisioner/distro/rhel-7.2.yml]: It's just a common information about the distro which is currently not used. But the distro yaml can be used by different apps. 

I can remove it but I think we would need that data later.

File: cli/spec.py [hunk 2]
```diff
@@ -130,6 +130,25 @@ def resolve_value(self, arg_name, defaults=None):
         self.value = utils.load_yaml(self.value, search_first)
 
 
+class SpecYamlArgument(YamlFileArgument):
```
Comment (yfried-redhat) [inline cli/spec.py]: This isn't needed at this moment. see comment about distro file

File: settings/provisioner/distro/rhel-7.2.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+image:
```
Comment (yfried-redhat) [inline settings/provisioner/distro/rhel-7.2.yml]: This file is specific for virsh, so please keep it in virsh settings

Comment (obaranov) [inline settings/provisioner/distro/rhel-7.2.yml]: Let's discuss that with @tkammer. In the ticket I had the requirement to have distro outside virsh provisioner. The idea was to have something similar to khaleese were we have distro yaml file with all the information about specific distro. And that infroamtion were used by different modules.

**Comment by yfried-redhat** [comment]
-1
Please don't merge this. let's discuss this in meeting

File: settings/provisioner/virsh/virsh.spec [hunk 4]
```diff
@@ -17,10 +17,10 @@ subparsers:
...
                       help: "User's SSH key"
                       default: ~/.ssh/id_rsa
-            - title: distro
+            - title: image
               options:
-                  distro:
-                      type: CommonSpecYamlFile
+                  image:
+                      type: YamlFile
                       help: The RHEL distribution version
```
Comment (yfried-redhat) [inline settings/provisioner/virsh/virsh.spec]: Help message is wrong here.

File: settings/provisioner/virsh/virsh.spec [hunk 5]
```diff
@@ -19,13 +19,9 @@ subparsers:
...
-                  image-file:
-                      type: Value
-                      help: An image to provision the host with
-                      required: yes
-                  image-server:
-                      type: Value
-                      help: Base URL of the image file server
+                  image:
+                      type: YamlFile
+                      help: The RHEL distribution version
```
Comment (yfried-redhat) [inline settings/provisioner/virsh/virsh.spec]: The help text is wrong here. please explain that this is the path to a file containing the image and the server details

**Comment by obaranov** [comment]
Fixed both public an private pull request.

**Pull request merged**